main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -313,6 +313,8 @@ object TypedAst {
 
     case class FixpointProject(pred: Name.Pred, exp: TypedAst.Expression, tpe: Type, pur: Type, eff: Type, loc: SourceLocation) extends TypedAst.Expression
 
+    case class Instanceof(exp: TypedAst.Expression, className: String, tpe: Type, pur: Type, eff: Type, loc: SourceLocation) extends TypedAst.Expression
```

Comment:
We can take a shortcut here:

tpe is just Bool, pur is just exp.pur, and eff is just exp.eff

Reply:
`Instanceof` constructors changed.